sources/models/src/modeled_types/shared.rs

```diff
@@ -326,8 +326,6 @@ mod test_version {
             "v1.0.1-alpha",
             "1.0.2-alpha+1.0",
             "v1.0.2-alpha+1.0",
-            "1.0.3-beta.1.01",
-            "v1.0.3-beta.1.01",
```

Comment:
These no longer parse? That seems odd.

Reply:
Yeah, it was a full replacement of the parser.  It used to depend on the semver-parser crate, which was a pest-based implementation by the same owner as semver, but semver was handed over to a new owner who improved performance and made some other structural changes.  The changes, sadly, weren't documented well, but the new version doesn't accept these versions as valid anymore.  Also sadly, different semver libraries for different languages don't agree on what's valid semver.  [This](https://jubianchi.github.io/semver-check/#/version/1.0.3-beta.1.01) says it's valid, for example.